usagov/vote-gov-drupal
# Vote-1651 ->  Updated external links test

## Jira ticket

[Vote-1651](https://bixal-projects.atlassian.net/browse/VOTE-1651)

## Description

This PR will update the external links test and make it into one file.  Before if we were to get a 403 we would then add it to a skip URL array and then add it to a separate file.  Now with this test, we are using both English and Spanish pages and checking based on the status of which test to run. 

- 200 -> run test
- 403 -> run excluded links test 
- 503 and 404 -> throw custom error 

Have also made this into one file and not multiple to help with maintenance. 

## Deployment and testing
### QA/Test

1. pull down branch 
2. start local 
3. cd into `testing`
4. run `lando drush uli` and log in as a super user, visit the Alabama state page, and update one of the links to be `https://vote.gov/help/` this will allow the test to check a 404 link (please save the link that you updated so that it can be fixed once testing is done.
5. return back to terminal 
6. run `npm run cy:open`
7. select the `external-links-validation.cy.js` and check that tests are passing, also check that Alabama is flagging a 404 and the error is showing, can also check Rhode Island which has a 403, and make sure that it is running the excluded links part of test
8. Go back to the Alabama state page and fix the link that was updated to the original one below is both links for easy reference. 

- `https://vote.gov/help/`
- `https://www.alabamainteractive.org/sos/voter_registration`

**note**
> Guam will fail but this is known and ticket has been made to fix

## Checklist for the Developer




## Checklist for the Peer Reviewers
